File: kernelSVM.py
# -*- coding: utf-8 -*-
"""
Created on Thu Jan  2 22:52:31 2020

@author: 138709
"""

# Kernel SVM Algorithm

# Importing the libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


# Import the dataset
dataset = pd.read_csv('Social_Network_Ads.csv')
X = dataset.iloc[:,[2,3]].values
Y = dataset.iloc[:,4].values


# Splitting the data into train and test sets
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.25, random_state = 0)


# Feature Scaling the data
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)


# Fitting Kernel SVM to the Training set
from sklearn.svm import SVC
model = SVC(kernel = 'rbf', random_state = 0)
model.fit(X_train, Y_train)


# Predicting the test set results
Y_pred = model.predict(X_test)


# Making the confusion matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(Y_test, Y_pred)

File: kernelSVM.R
# Kernel - SVM
install.packages('caTools')
library('caTools')

# Import the dataset
dataset <- read.csv('Social_Network_Ads.csv')
dataset = dataset[3:5]


# Encoding the target feature as factor
dataset$Purchased = factor(dataset$Purchased, levels = c(0,1))


# Splitting the dataset into training and test set
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.75)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


# Feature Scaling
training_set[-3] = scale(training_set[-3])
test_set[-3] = scale(test_set[-3])


# Fitting Kernel SVM to the training set
install.packages('e1071')
library(e1071)
model = svm(formula = Purchased ~ ., data = training_set, type = 'C-classification', kernel = 'radial')



# Predicting the test set results
Y_pred = predict(model, newdata = test_set[-3])



# Making the confusion matrix
cm = table(test_set[ ,3], Y_pred)


